lib/Conversion/ATenToTCF/CoreOpConversionPatterns.cpp

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "npcomp/Conversion/ATenToTCF/Patterns.h"
+
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/PatternMatch.h"
+#include "npcomp/Dialect/ATen/IR/ATenDialect.h"
+#include "npcomp/Dialect/TCF/IR/TCFOps.h"
+
+using namespace mlir;
+using namespace mlir::NPCOMP;
+
+namespace {
+
+/// The ATen AddOp actually has three arguments:
+///   self, other, alpha
+/// Alpha is an integer that is multiplied by 'other' prior to adding.
+class ConvertATenAdd : public OpRewritePattern<aten::AddOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(aten::AddOp srcAddOp,
+                                PatternRewriter &rewriter) const override {
+    // Special case: Match when alpha is constant 1, which is the default,
+    // quite common and maps directly to a TCF add.
+    APInt alphaValue;
+    if (matchPattern(srcAddOp.alpha(), m_ConstantInt(&alphaValue)) &&
```

Comment:
Is it intended that alpha is an integer here? I thought it is usually a floating point.


https://pytorch.org/docs/stable/generated/torch.add.html

"If other is of type FloatTensor or DoubleTensor, alpha must be a real number, otherwise it should be an integer."

Reply:
The docs are quite aspirational on this score, and as near as I can tell: a) it isn't enforced and normal type promotion rules apply (just like with all other arithmetic), and b) when defaulted, it defaults to a const 1 (integer) with no further checking. For this early phase, I am more keying off of the default value case since we lack the op coverage to do the generic case with type promotion. I've clarified the comment.